src/pages/docs/guides/chat/build-livestream.mdx

```diff
@@ -0,0 +1,240 @@
+
+The most important decision you can make is the experience you want users to have. It determines the look and feel of the livestream and ultimately the impression your users leave with. The most important architectural decision therefore is how you structure your chat rooms. With Ably, you are not limited by technology - only by the user experience you want to deliver. With Ably you can deliver experiences that are:
+
+- **Single massive room:** Ably seamlessly supports millions of users in a single room, by automatically distributing websocket connections across multiple servers to ensure endless connection scalability. This is ideal for global events or streams where everyone participates together.
+- **Many concurrent rooms:** You can run thousands of rooms in parallel, each with their own audience and message flow. Not only do connections scale horizontally, but Ably uses [consistent hashing](https://ably.com/blog/implementing-efficient-consistent-hashing) to evenly distribute the management of chat rooms across instances, allowing the number of rooms to grow without limit and without causing load spikes when you do.
+
+PLACEHOLDER: Diagram comparing single-room vs. multi-room architectures
+
+In making this decision, there are some **key considerations**:
+
+- Should everyone see the same messages, or do you need to segment by language, region, friend group or topic?
+  - If you're running livestreams over multiple languages or geographical regions, you may want to consider separating users similarly so that everybody understands the chat.
+  - Alternatively, you may want to make use of translation services, that can translate messages on the fly to the language of the user.
+- Do you want an environment where everyone has time to read every message, or the chat to go wild with hype when something exciting happens?
+  - At anything more than 5-10 messages per second, the chat becomes unreadable for the audience.
+  - **Sharding** is one approach to this problem - separating users into smaller groups so that the chat is more manageable. This can be done randomly, or using some predefined scheme, e.g. language, geographical region.
+  - You can also simply _drop_  a proportion of messages on the client to keep it readable. If you're not worried about everyone seeing everything, this is a simple approach.
+  - Of course, you can also rate-limit how often users send messages. This controls the rate of messages, but can negatively impact the user experience if they're not able to respond when they want to.
+- Do you need to support both public and private chat? If you do, you might want one big room for public chat and a separate room for private messages.
+
+Ably gives you the flexibility to design for your users, not your infrastructure.
+
+## Message throughput and rate management
+
+Livestream chat is defined by high message rates and unpredictable spikes. Ably is engineered to handle these loads without degradation.
```

Comment:
Good place to link to more architectural content backing this up.

Reply:
Done